### PR TITLE
Submission to Avalanche Tutorial Contest - Precompiled Storage Gateway

### PR DESCRIPTION
**Pull request by modenero**
This tutorial is currently a work-in-progress. However, I do hope that it provides an adequate demonstration of the potential for educating and growing the Subnet community.

Following the guidelines expressed to other participants, special attention has been made to:
1. Verify that the tutorial is building correctly.
2. All mentions of Subnet are capitalized
3. Good use of Markdown features (ie. tables, annotations, links, images, etc)

Greatly appreciate this opportunity to contribute to the Avalanche (Subnet) ecosystem.

Cheers!
Shomari

### CODE REVIEW

**Comment by jpop32** [comment]
Unfortunately, new entries were only accepted in Round 1, which had a deadline of July 6th. Watch out for new contests coming up.

**Comment by modenero** [comment]
@jpop32 @cgcardona 

I've reached back out to Karina Popovich for guidance, but as far as I know, I was invited (on July 14th) to submit my tutorials in Round 2.

Please advise if this tutorial (along with the 2 others I submitted) would still be considered for inclusion in the ongoing contest. Also I've continued to develop all of these tutorials over the past week and I'd also really appreciate an opportunity to update to their latest revisions.

Thank you,
Shomari

**Comment by jpop32** [comment]
Apologies, this submission **was** selected for R2.

File: docs/community/tutorials-contest/2022/precompile-storage-gateway/README.md [hunk 1]
```diff
@@ -0,0 +1,1222 @@
...
+Well done, you made it!  
+💯 🔥 👏 🎊
+
+There are countless decentralized applications that can utilize the convenience of external storage providers directly accessible to their smart contracts.
+
+### Recap of what we learned
+
+- How to build `subnet-evm` from [source](https://github.com/ava-labs/subnet-evm) with a custom [Precompile](/subnets/customize-a-subnet#precompiles) contract
+- How to deploy a [Solidity](https://docs.soliditylang.org/) contract using the [Remix IDE](http://remix.ethereum.org/)
+- How to interact with a [Precompile](/subnets/customize-a-subnet#precompiles) contract using the [Ethers.js](https://docs.ethers.io/v5/) library
```
Comment (Julian-dev28) [inline docs/community/tutorials-contest/2022/precompile-storage-gateway/README.md:1186]:
```suggestion
- How to interact with a [precompiled](/subnets/customize-a-subnet#precompiles) contract using the [Ethers.js](https://docs.ethers.io/v5/) library
```

**Review by Julian-dev28 — CHANGES_REQUESTED** [review]

File: docs/community/tutorials-contest/2022/precompile-storage-gateway/README.md [hunk 2]
```diff
@@ -0,0 +1,1222 @@
...
+The following are a few of the most common issues faced by the Subnet developer community.
+
+### Troubleshooting
+
+There are still issues that need to be solved, related to:
+
+1. Contract permissions
+2. Request timeout
+
+#### Contract permissions
```
Comment (yulin-dong) [inline docs/community/tutorials-contest/2022/precompile-storage-gateway/README.md:1157]: All section header should be title case, please check and update all section headers with https://titlecase.com/

**Review by yulin-dong — CHANGES_REQUESTED** [review]

File: docs/community/tutorials-contest/2022/precompile-storage-gateway/README.md [hunk 3]
```diff
@@ -0,0 +1,1222 @@
...
+
+:::info
+
+__Did you know? —__ You can easily customize the user permissions, gas and precompiled contracts for your Subnet using the new Avalanche CLI.
+
+:::
+
+### Supported networks
+
+Allows Avalanche to interact with several decentralized storage systems that can hold a bigger data load that the on-chain, for example:
```
[the remark this answers is not in the document]
Comment (modenero) [reply docs/community/tutorials-contest/2022/precompile-storage-gateway/README.md]: ewww, not sure what I was trying to say there .. fixed!

File: docs/community/tutorials-contest/2022/precompile-storage-gateway/README.md [hunk 4]
```diff
@@ -0,0 +1,1222 @@
...
+git clone https://github.com/ava-labs/subnet-evm.git
+```
+
+2. Modify `params/config.go`
+
+```bash
+vim params/config.go
+```
+
+- update these objects ↴
```
[the remark this answers is not in the document]
Comment (modenero) [reply docs/community/tutorials-contest/2022/precompile-storage-gateway/README.md]: I've indicated the approx line number(s) for code insertion

plus added the actual line numbers to each of the code blocks

File: docs/community/tutorials-contest/2022/precompile-storage-gateway/README.md (hunk 2 above)
Comment (modenero) [reply docs/community/tutorials-contest/2022/precompile-storage-gateway/README.md]: done!

**Comment by modenero** [comment]
@Julian-dev28 
@yulin-dong 

Please forgive me as Git is not my first language. I didn't realize that the "forced" updates to my own repo were in fact being reflected here as well. I'll be more careful in the future when making updates to my own repo.

First off, thank you so much selecting this tutorial for review in your competition 🙏 This latest update is pretty close to the version that I wish had been properly submitted last Friday. 

The only concern I have is regarding the Walk-through Video that I've placed at the end of the Introduction section. My intention here is to summarize the key points of the tutorial in a short 5-7 minute video (with a 20-30 second gif preview). I'm also considering adding additional walk-through videos to detail each individual key point (at their respective points in the tutorial).

- So what's your opinion on videos? 
- would you just prefer static PNG screenshots? 
- or perhaps GIF animated screenshots?

Once I get a sense of what the Ava Team is looking for here, I'll get to work immediately on finalizing the graphic & multimedia elements of the tutorial.

Very much looking forward to your continued feedback; and apologize for the delay in starting this review process.

Enjoy the rest of your weekend! 🏖️ 🍹 😎  
Shomari

**Comment by yulin-dong** [comment]
Please fix broken build

**Comment by modenero** [comment]
@yulin-dong 

I've run the build & serve on the docs using `node v16.15.0` and it completes successfully

```
yarn build
yarn run v1.22.19
$ docusaurus build

   ╭──────────────────────────────────────────────────────────────────────────────────────────╮
   │                                                                                          │
   │                       Update available 2.0.0-beta.22 → 2.0.0-rc.1                        │
   │                                                                                          │
   │    To upgrade Docusaurus packages with the latest version, run the following command:    │
   │     `yarn upgrade @docusaurus/core@latest @docusaurus/plugin-google-analytics@latest     │
   │        @docusaurus/preset-classic@latest @docusaurus/theme-search-algolia@latest`        │
   │                                                                                          │
   ╰──────────────────────────────────────────────────────────────────────────────────────────╯

[INFO] [en] Creating an optimized production build...

✔ Client
  

✔ Server
  Compiled successfully in 23.00s


✔ Client
  

● Server █████████████████████████ cache (99%) shutdown IdleFileCachePlugin
 stored

[SUCCESS] Generated static files in "build".
[INFO] Use `npm run serve` command to test your build locally.
Done in 29.48s.
```

I don't understand the error with the CI

```
Please check the pages of your site in the list below, and make sure you don't reference any path that does not exist.
Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.

Exhaustive list of all broken links found:

- On source page path = /community/tutorials-contest/2022:
   -> linking to entry_directory/README.md (resolved as: /community/tutorials-contest/entry_directory/README.md)
```

Do you have any idea what could be causing this error? 
Or how to resolve it?

**Comment by modenero** [comment]
@yulin-dong 

I'm seeing "Warnings" in the logs as well

```
Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/[20](https://github.com/ava-labs/avalanche-docs/runs/7596310438?check_suite_focus=true#step:5:21)22/README.md" for version current
Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/20[22](https://github.com/ava-labs/avalanche-docs/runs/7596310438?check_suite_focus=true#step:5:23)/README.md" for version current
```

it lists the `avalanche-docs` directory twice in the path

```
/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md
```

perhaps this is related to the build error??

**Comment by Julian-dev28** [comment]
> @Julian-dev28 @yulin-dong
> 
> Please forgive me as Git is not my first language. I didn't realize that the "forced" updates to my own repo were in fact being reflected here as well. I'll be more careful in the future when making updates to my own repo.
> 
> First off, thank you so much selecting this tutorial for review in your competition 🙏 This latest update is pretty close to the version that I wish had been properly submitted last Friday.
> 
> The only concern I have is regarding the Walk-through Video that I've placed at the end of the Introduction section. My intention here is to summarize the key points of the tutorial in a short 5-7 minute video (with a 20-30 second gif preview). I'm also considering adding additional walk-through videos to detail each individual key point (at their respective points in the tutorial).
> 
> * So what's your opinion on videos?
> * would you just prefer static PNG screenshots?
> * or perhaps GIF animated screenshots?
> 
> Once I get a sense of what the Ava Team is looking for here, I'll get to work immediately on finalizing the graphic & multimedia elements of the tutorial.
> 
> Very much looking forward to your continued feedback; and apologize for the delay in starting this review process.
> 
> Enjoy the rest of your weekend! 🏖️ 🍹 😎 Shomari

Static Images are fine

**Comment by Julian-dev28** [comment]
> 

Can you please ensure that all links work

**Comment by yulin-dong** [comment]
> @yulin-dong
> 
> I'm seeing "Warnings" in the logs as well
> 
> ```
> Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
> Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
> Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
> Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
> Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
> Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
> Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
> Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
> Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
> Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/[20](https://github.com/ava-labs/avalanche-docs/runs/7596310438?check_suite_focus=true#step:5:21)22/README.md" for version current
> Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md" for version current
> Warning:  Docs markdown link couldn't be resolved: (entry_directory/README.md) in "/home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/20[22](https://github.com/ava-labs/avalanche-docs/runs/7596310438?check_suite_focus=true#step:5:23)/README.md" for version current
> ```
> 
> it lists the `avalanche-docs` directory twice in the path
> 
> ```
> /home/runner/work/avalanche-docs/avalanche-docs/docs/community/tutorials-contest/2022/README.md
> ```
> 
> perhaps this is related to the build error??

Please pull/merge and rebuild. It should work now.

**Comment by modenero** [comment]
> Static Images are fine

ok thanks. i'll have the static images/screenshots added by end-of-day.

**Comment by modenero** [comment]
@yulin-dong 

As the winners have already been announced, I suppose I'll just close this now.

Very disappointed with this outcome, but I appreciate the opportunity nonetheless.

All the best to you and the rest of your team.

Cheers!
Shomari

**Pull request closed without merging**